packages/pigeon/bin/pigeon.dart

```diff
@@ -4,7 +4,7 @@
 
 import 'dart:io' show exit;
 
-import 'package:pigeon/pigeon_cl.dart';
+import 'package:pigeon/src/pigeon_cl.dart';
```

Comment:
Interesting, I guess `bin/` has special-case allowance in the analyzer to import from the same package's `src` even if it's using a `package:` import?

Reply:
I was surprised it was allowed as well, since the pigeon_cl.dart file is used outside of lib and tests, should I move it back out of src?

Reply:
Nope, I found a reference later on the page I had linked to, it's explicitly allowed:

> You are free to import libraries that live in lib/src from within other Dart code in the same package (like other libraries in lib, **scripts in bin**, and tests) but you should never import from another package's lib/src directory.

And that's the preferred format for such an include:

> How you import libraries from within your own package depends on the locations of the libraries:
> 
> When [reaching inside or outside lib/](https://dart.dev/effective-dart/usage#dont-allow-an-import-path-to-reach-into-or-out-of-lib) (lint: [avoid_relative_lib_imports](https://dart.dev/tools/linter-rules/avoid_relative_lib_imports)), use package:.